lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -69,6 +70,90 @@ export class PublicClientApplication extends ClientApplication implements IPubli
         return this.acquireTokenPopup(request || DEFAULT_REQUEST);
     }
 
+    /**
+     * This function uses a hidden iframe to fetch an authorization code from the eSTS. There are cases where this may not work:
+     * - Any browser using a form of Intelligent Tracking Prevention
+     * - If there is not an established session with the service
+     *
+     * In these cases, the request must be done inside a popup or full frame redirect.
+     *
+     * For the cases where interaction is required, you cannot send a request with prompt=none.
+     *
+     * If your refresh token has expired, you can use this function to fetch a new set of tokens silently as long as
+     * you session on the server still exists.
+     * @param request {@link SsoSilentRequest}
+     *
+     * @returns A promise that is fulfilled when this function has completed, or rejected if an error was raised.
+     */
+    async ssoSilent(request: SsoSilentRequest): Promise<AuthenticationResult> {
+        this.preflightBrowserEnvironmentCheck(InteractionType.Silent);
+        this.emitEvent(EventType.SSO_SILENT_START, InteractionType.Silent, request);
+
+        try {
+            // Default skipCache to true, if not passed. 
+            const {
+                skipCache = true
+            } = request;
+
+            // Build cached account by loginHint, sid, or the request itself
+            const accountByLoginHint = request.loginHint && this.getAccountByUsername(request.loginHint);
+            
+            const accountBySid = request.sid && this.getAccountBySessionId(request.sid);
+            if (request.sid && !accountBySid && !skipCache) {
+                this.logger.warning("ssoSilent - sid provided to ssoSilent, but no cached account found. If this is unexpected, confirm sid is enabled as an optional id token claim for your application");
+            }
+
+            const account = request.account || accountBySid || accountByLoginHint;
+
+            // Only checked for cached credentials if forceRefresh is off and there is an account for the request
+            if (!skipCache && account) {
+                const silentRequest: SilentRequest = {
+                    ...request,
+                    account,
+                    scopes: request.scopes || DEFAULT_REQUEST.scopes
+                };
+
+                this.logger.verbose("ssoSilent - checking for cached tokens.");
+
+                const silentResponse = await this.acquireTokenSilent(silentRequest)
+                    .catch(error => {
+                        /*
+                         * This error is returned when there is an empty cache. 
+                         * In this scenario, fall through to iframed request.
+                         */
+                        if (error.errorCode === ClientAuthErrorMessage.noTokensFoundError.code) {
+                            return null;
+                        }
+
+                        throw error;
```

Comment:
Might have to take another look at this later for telemetry purposes (it can be an incremental update)